base/sort.jl

```diff
@@ -426,7 +426,7 @@ function selectperm!{I<:Integer}(ix::AbstractVector{I}, v::AbstractVector,
                                  order::Ordering=Forward,
                                  initialized::Bool=false)
     if !initialized
-        @inbounds for i = 1:length(ix)
+        @inbounds for i in eachindex(ix)
```

Comment:
I am slightly uneasy about all this from a performance point of view. From what I have seen before, it is quite easy to accidentally introduce a 20-30% regression by causing something to not inline anymore. I know you have done some benchmarks with zip etc but are they valid in all cases that are being changed in all these PRs? To me, something like this would be better in a separate branch and then running the benchmark suite on the whole thing,